arrow_pd_parser/export.py

```diff
@@ -67,3 +71,32 @@ def pd_to_json(
             new[col].replace("NaT", np.nan, regex=False, inplace=True)
 
     new.to_json(output_file, orient=orient, lines=lines, **kwargs)
+
+
+def pd_to_parquet(
+    df: pd.DataFrame,
+    output_file: str,
+    from_pandas_kwargs={},
+    write_table_kwargs={},
+    arrow_schema: pa.lib.Schema = None,
+):
+    """
+    Export a data frame as parquet
+    Does no conversion, and adheres to the parquet schema
+
+    Args:
+        df (pd.DataFrame): a pandas dataframe
+        output_file str: the path you want to export to (s3)
+    """
+
+    if not isinstance(output_file, str):
+        raise TypeError("currently only supports string paths for output")
```

Comment:
Can't remember why this was an issue?

Reply:
`pyarrow.parquet.write_table` only accepts `str` or `pyarrow.lib.nativefile` types as the output destination not the normal IO file type that the pandas `df.to_parquet' accepts. the pyarrow method was chosen so a schema could be applied on output to parquet. so the following could be allowed:
- allow the use of a `pyarrow.lib.nativefile` (returned by pyarrow fs)
- use df.to_parquet (but don't allow schema)
- continue to only allow str paths

Reply:
Interesting, might be worth seeing what Pandas does under the hood? As if you did `df.to_parquet()` uses pyarrow. So if you gave it a `BytesIO` object and it worked. It would be interesting to see how pandas forces pyarrow to use it (although diving into the read parts of Pandas might not be that quick).

I would definitely go with option 1 for now as it is the quickest. As it is just a wrapper function for pyarrow I think it is fine to give it a `pyarrow.lib.nativefile` or `str` for parquet read/write only.

Reply:
it only works for write - it refuses to read using this method (even though providing `pyarrow.lib.nativefile` is supposed to be ok [as said here](http://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html) 
I have allowed the write to accept a `pyarrow.lib.nativefile` and put the type error in for reading